src/net/sourceforge/kolmafia/maximizer/CheckedItem.java

```diff
@@ -229,6 +229,14 @@ private static int limitBuyable(final int itemId) {
       case ItemPool.REFLEX_HAMMER:
       case ItemPool.MAYO_LANCE:
         return Preferences.getBoolean("_mayoDeviceRented") ? 0 : 1;
+      case ItemPool.FEDORA_MOUNTED_FOUNTAIN:
```

Comment:
Perhaps we could use `NPCPurchaseRequest.getQuantity()` here (and move over the mayo stuff?) to reduce this sort of duplication

Reply:
Done, but I put it in `NPCStoreDatabase`, which has yet another, presumably slightly different, method.

The change in `RequestProcessor` is to fix what I think has happened in some of my runs: if you buy the hat using e.g. `/buy fedora-mounted fountain` Mafia doesn't load fwshop and so doesn't record the purchase.